rust/agents/relayer/src/relayer.rs

```diff
@@ -1,24 +1,27 @@
+use abacus_core::MultisigSignedCheckpoint;
 use async_trait::async_trait;
 use eyre::{Context, Result};
-use tokio::task::JoinHandle;
+use tokio::{
+    sync::watch::{channel, Receiver, Sender},
+    task::JoinHandle,
+};
+
 use tracing::{instrument::Instrumented, Instrument};
 
 use abacus_base::{
     AbacusAgentCore, Agent, ContractSyncMetrics, InboxContracts, MultisigCheckpointSyncer,
 };
 
 use crate::{
-    checkpoint_relayer::CheckpointRelayer, message_processor::MessageProcessor,
+    checkpoint_fetcher::CheckpointFetcher, message_processor::MessageProcessor,
     settings::RelayerSettings as Settings,
 };
 
 /// A relayer agent
 #[derive(Debug)]
 pub struct Relayer {
-    polling_interval: u64,
-    max_retries: u32,
-    submission_latency: u64,
-    relayer_message_processing: bool,
+    signed_checkpoint_polling_interval: u64,
```

Comment:
Not really worth worrying about for this PR, but honestly, I think overly verbose names are a bit overkill; they just need to be clear enough. We can and _should_ make use of doc comments to elaborate further on the specifics which will show up on hover in any decent IDE.